### DISCUSSION TITLE
[Soketi Serverless] Horizontal scale won't work properly

### DISCUSSION THREAD

**Discussion by Tiagospem** [soketi/pws#733] [Q&A] [answered]
When using the soketi with the horizontal scale option, the communication between the clients is closed after 3 minutes of inactivity, no event is received by the clients, however the client continues to be pinged and receiving. 

<img width="482" alt="CleanShot 2022-10-18 at 16 17 43@2x" src="https://user-images.githubusercontent.com/8459781/196523737-bd899deb-bcfb-46d1-9bab-d8d36de99593.png">

When configuring the worker on single shard, no issues are encountered.

**Comment by NAB5** [comment]
Seeing the same behavior as well.

**Comment by rennokki** [comment]
@Tiagospem Please use the bug template to pass more details.

**Comment by Tiagospem** [comment]
@rennokki 



**Description**
On create a Cloudflare Worker using 'Horizontally Scalable Clustering' on dash.soketi.app this won't work properly, it stops delivery events 3-4 minutes after page load, no erros are displayed on console. Using single-instance mode, works fine.

**Reproduction steps**
Steps to reproduce the behavior:

1. create a fresh install of laravel
2. create a simple event using a presencial channel
3. create a cloudflare worker using horizontally scale option
4. create a new client on soketi dashboard app.
5. setup your credentials in laravel .env
6. create a view and update your app.js to listen for the event your created before
7. build your app, run migrations, create a new user to use presencial channel
8. open the app in browser and send an event, it should work.
9. wait some minutes like 3-4 minutes and send event again, nothing should happen.

**Expected behavior**
I expect the connection to remain open as long as the user has the app open, and events are delivered, just like the single-instance clustering option.

**Screenshots**
Here you can see open communication, events are not delivered. within 3 minutes, it simply stops working.

<img width="482" alt="196523737-bd899deb-bcfb-46d1-9bab-d8d36de99593" src="https://user-images.githubusercontent.com/8459781/199983069-45ff8529-f064-472e-8c73-f7cb9d7b306d.png">


**Environment**
- laravel/framework: ^9.19
- Adapter: pusher

**Configuration**
```js
   window.Echo = new Echo({
    broadcaster: 'pusher',
    key: import.meta.env.VITE_PUSHER_APP_KEY,
    wsHost: import.meta.env.VITE_PUSHER_APP_HOST,
    wsPort: import.meta.env.VITE_PUSHER_APP_PORT,
    wssPort: import.meta.env.VITE_PUSHER_APP_PORT,
    forceTLS: import.meta.env.VITE_PUSHER_APP_PORT === 443,
    encrypted: true,
    disableStats: true,
    enabledTransports: ['ws', 'wss'],
});
```

**Comment by romamd** [comment]
@rennokki Hi, I recently started to use soketi, thanks for an awesome library!
I was trying to use horizontally scalable worker on cloudflare, but I have the same issue, just with a much simpler scenario:
- create a cloudflare worker using horizontally scale option
- create a client to listen for the events (used code provided in the dashboard)
- send the events, after 3-4 minutes of inactivity, send an event again, observe it's not received by client.

I also observed that if I redeploy the worker from cloudflare, then send an event again, client receives events. And then, after an inactivity interval, it starts to reproduce again.

Is there any option to deploy soketi manually to cloudflare? I would like to help debug and fixing this

**Comment by rennokki** [comment, marked as answer]
Upgrade your Workers to `0.7.0`. This fixes the issues. 👍🏼 

**Make sure to read the release notes when upgrading in the dashboard.**
